docs/L1_L2_Messages.md

```diff
@@ -0,0 +1,233 @@
+
+### L2 to L1 Messages Lifecycle && API
+
+The lifecycle of sending a message from layer 2 to layer 1 can be broken down into roughly 4 steps, only 2 which (at most!) require the end user to publish transactions.
+
+**1. Publish L2 to L1 transaction (Arbitrum transaction)**
+
+A client initiates the process by publishing a message via `ArbSys.sendTxToL1` (see [ArbSys](Arbsys.md), and see [ArbTokenBridge.\_withdraw](https://github.com/OffchainLabs/arbitrum/blob/5bd9a456a780582715a62affd887d35e2eb138b0/packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/ArbTokenBridge.sol#L256) for example usage.
+
+**2. Outbox entry gets created**
+
+After the Arbitrum chain advances some set amount of time, ArbOS gathers all outgoing messages, Merklizes them, and publishes the root as an [OutboxEntry](https://github.com/OffchainLabs/arbitrum/blob/master/packages/arb-bridge-eth/contracts/bridge/OutboxEntry.sol) in the chain's outbox. Note that this happens "automatically"; i.e., it requires no additional action from the user.
+
+**3. Client gets Merkle proof of outgoing message**
+
+After the Outbox entry is published on chain, the user (or anybody) can compute the Merkle proof of inclusion of their outgoing message. They do this by calling `NodeInterface.lookupMessageBatchProof`:
+
+```sol
+interface NodeInterface {
+    /**
+    * @notice Returns the proof necessary to redeem a message
+    * @param batchNum index of outbox entry (i.e., outgoing messages Merkle root) in array of outbox entries
+    * @param index index of outgoing message in outbox entry
+    * @return (
+        * proof: Merkle proof of message inclusion in outbox entry
+        * path: Index of message in outbox entry
+        * l2Sender: sender if original message (i.e., caller of ArbSys.sendTxToL1)
+        * l1Dest: destination address for L1 contract call
+        * l2Block l2 block number at which sendTxToL1 call was made
+        * l1Block l1 block number at which sendTxToL1 call was made
+        * timestamp l2 Timestamp at which sendTxToL1 call was made
+        * amouunt value in L1 message in wei
+        * calldataForL1 abi-encoded L1 message data
+        *
+    */
+    function lookupMessageBatchProof(uint256 batchNum, uint64 index)
+        external
+        view
+        returns (
+            bytes32[] memory proof,
+            uint256 path,
+            address l2Sender,
+            address l1Dest,
+            uint256 l2Block,
+            uint256 l1Block,
+            uint256 timestamp,
+            uint256 amount,
+            bytes memory calldataForL1
+        );
+}
+
+```
+
+**4. The user executes the L1 message (Ethereum Transaction)**
+
+Anytime after the dispute window passes, a user can execute the L1 message by calling Outbox.executeTransaction:
```

Comment:
We should probably mention that, anyone can execute and if the tx reverts it is re-executable - *forever* .